### PR TITLE
Send SMS - additional number format handling & code clean-up

### PR DESCRIPTION
**Pull request by tahir-khalid**
## Description

Sentry reporting an error: InvalidPhoneError: Not a UK mobile number (Notifications::Client::BadRequestError)

Added additional number format handling and clean up and updated spec to match.

### CODE REVIEW

File: app/services/send_sms.rb [hunk 1]
```diff
@@ -33,22 +33,21 @@ def otp_code(mobile_number:, code:)
   def validate_and_format_phone_number(phone_number)
     return nil if phone_number.nil?
 
-    normalized_number = normalize_uk_number(phone_number)
+    normalized_number = normalize_uk_number(phone_number.gsub(/[\s-]+/, ""))
     return nil unless valid_phone_number?(normalized_number)
 
-    Phonelib.parse(normalized_number).international
+    Phonelib.parse(normalized_number).international.gsub(/[\s-]+/, "")
```
Comment (meeravnagesh) [inline app/services/send_sms.rb:39]: the input to this is already been cleansed and you have removed the space and -

File: app/services/send_sms.rb [hunk 2]
```diff
@@ -33,22 +33,21 @@ def otp_code(mobile_number:, code:)
...
   end
 
   def normalize_uk_number(phone_number)
-    return phone_number if phone_number.start_with?(UK_PREFIX)
+    cleaned_number = phone_number.gsub(/[\s-]+/, "")
```
Comment (meeravnagesh) [inline app/services/send_sms.rb:43]: the phone_numer parameter is already gsubed

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: app/services/send_sms.rb [hunk 3]
```diff
@@ -36,19 +36,18 @@ def validate_and_format_phone_number(phone_number)
     normalized_number = normalize_uk_number(phone_number)
     return nil unless valid_phone_number?(normalized_number)
 
-    Phonelib.parse(normalized_number).international
+    Phonelib.parse(normalized_number).international.gsub(/[\s-]+/, "")
```
Comment (meeravnagesh) [inline app/services/send_sms.rb:39]: image

Comment (meeravnagesh) [reply app/services/send_sms.rb:39]: here the code in normalise UK number is removing spaces and in the end the Parse.international is adding spaces and then again your gsub is removing spaces

Comment (meeravnagesh) [reply app/services/send_sms.rb:39]: remove this line - its not needed anymore if you dont want spaces

File: app/services/send_sms.rb [hunk 4]
```diff
@@ -36,19 +36,18 @@ def validate_and_format_phone_number(phone_number)
...
+    if cleaned_number.match?(UK_LEADING_ZERO)
+      return UK_PREFIX + cleaned_number.gsub(UK_LEADING_ZERO, "")
     end
 
-    # If number doesn't start with + or 0, assume it needs +44
-    UK_PREFIX + phone_number
+    UK_PREFIX + cleaned_number
   end
 
   def valid_phone_number?(phone_number)
```
Comment (meeravnagesh) [inline app/services/send_sms.rb:53]: for your new case/requirement , this below code reckon below code is enough
private

  def validate_and_format_phone_number(phone_number)
    return nil if phone_number.nil?

    normalized_number = normalize_uk_number(phone_number)
    return nil unless valid_phone_number?(normalized_number)
    normalized_number
  end

  def normalize_uk_number(phone_number)
    cleaned_number = phone_number.gsub(/[\s-]+/, "")
    return cleaned_number if cleaned_number.start_with?(UK_PREFIX)

    if cleaned_number.match?(UK_LEADING_ZERO)
      return UK_PREFIX + cleaned_number.gsub(UK_LEADING_ZERO, "")
    end

    UK_PREFIX + cleaned_number
  end

  def valid_phone_number?(phone_number)
    Phonelib.parse(phone_number).valid?
  end
end

Comment (meeravnagesh) [reply app/services/send_sms.rb:53]: in normalised the UK number and added a prefix +44, no need to check again
parsed_number.valid? && parsed_number.country_code == "44" # Ensure it's a UK number

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: app/services/send_sms.rb (hunk 3 above)
Comment (tahir-khalid) [reply app/services/send_sms.rb:39]: Refactored.

File: app/services/send_sms.rb (hunk 4 above)
Comment (tahir-khalid) [reply app/services/send_sms.rb:53]: Implemented.

**Review by meeravnagesh — APPROVED** [review]

**Pull request merged**